File: LEETCODE/680. 验证回文字符串 Ⅱ.java
class Solution {
    public boolean validPalindrome(String s) {
        int i=0,j=s.length()-1;
        while(i<j){
            //不相等的话就不是回文啦！然后就删除i或者j判断是不是回文就好了！有一个是就够了！
            if(s.charAt(i)!=s.charAt(j)){
                return isPalindrome(s,i+1,j)||isPalindrome(s,i,j-1);
            }else{//相等的话就继续往中间查询
                i++;
                j--;
            }
        }
        return true;
        
    }
    
    //判断一个字符串是不是回文串
    private boolean isPalindrome(String s,int i,int j){
        while(i<j){
            if(s.charAt(i++)!=s.charAt(j--)){
                return false;
            }
        }
        return true;
    }
}

File: LEETCODE/kotlin/680. 验证回文字符串 Ⅱ.kt
class Solution {
    fun validPalindrome(s: String): Boolean {
        var i=0
        var j=s.length-1
        while(i<j){
            if(s[i]!=s[j]){
                return isPalindrome(s, i+1, j) || isPalindrome(s, i, j-1)
            }
            i++
            j--
        }
        return true
    }
    fun isPalindrome(s: String, ii: Int, jj: Int): Boolean {
        var i = ii
        var j = jj
        while(i<j){
            if(s[i]==s[j]){
                i++
                j--
            }else{
                return false
            }
        }
        return true
    }
}